aws/resource_aws_api_gateway_vpc_link_test.go

```diff
@@ -0,0 +1,144 @@
+					resource.TestCheckResourceAttr("aws_api_gateway_vpc_link.test", "description", "test"),
+				),
+			},
+			{
+				Config: testAccAPIGatewayVpcLinkConfig_Update(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsAPIGatewayVpcLinkExists("aws_api_gateway_vpc_link.test"),
+					resource.TestCheckResourceAttr("aws_api_gateway_vpc_link.test", "name", fmt.Sprintf("tf-apigateway-update-%s", rName)),
+					resource.TestCheckResourceAttr("aws_api_gateway_vpc_link.test", "description", "test update"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckAwsAPIGatewayVpcLinkDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).apigateway
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_api_gateway_vpc_link" {
+			continue
+		}
+
+		input := &apigateway.GetVpcLinkInput{
+			VpcLinkId: aws.String(rs.Primary.ID),
+		}
+
+		resp, err := conn.GetVpcLink(input)
+		if err != nil {
+			if ecrerr, ok := err.(awserr.Error); ok {
+				switch ecrerr.Code() {
+				case apigateway.ErrCodeNotFoundException:
+					return nil
+				}
+			}
+			return err
+		}
+
+		if *resp.Status != apigateway.VpcLinkStatusDeleting {
+			return fmt.Errorf("APIGateway VPC Link (%s) not deleted", rs.Primary.ID)
+		}
+	}
+
+	return nil
+}
+
+func testAccCheckAwsAPIGatewayVpcLinkExists(name string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		_, ok := s.RootModule().Resources[name]
+		if !ok {
+			return fmt.Errorf("Not found: %s", name)
+		}
+
+		return nil
+	}
+}
+
+func testAccAPIGatewayVpcLinkConfig(rName string) string {
+	return fmt.Sprintf(`
+resource "aws_lb" "test_a" {
+  name = "tf-lb-a-%s"
+  internal = true
+  load_balancer_type = "network"
+
+  subnet_mapping {
+    subnet_id = "${aws_subnet.test.id}"
+  }
+}
+
+resource "aws_vpc" "test" {
+  cidr_block = "10.10.0.0/16"
+}
+
+resource "aws_subnet" "test" {
+  vpc_id = "${aws_vpc.test.id}"
+  cidr_block = "10.10.0.0/21"
+  map_public_ip_on_launch = true
+  availability_zone = "us-west-2a"
+}
+
+resource "aws_api_gateway_vpc_link" "test" {
+  name = "tf-apigateway-%s"
+  description = "test"
+  target_arn = "${aws_lb.test_a.arn}"
+}
+`, rName, rName)
+}
+
+func testAccAPIGatewayVpcLinkConfig_Update(rName string) string {
+	return fmt.Sprintf(`
+resource "aws_lb" "test_a" {
+  name = "tf-lb-a-%s"
+  internal = true
+  load_balancer_type = "network"
+
+  subnet_mapping {
+    subnet_id = "${aws_subnet.test.id}"
+  }
+}
+
+resource "aws_vpc" "test" {
+  cidr_block = "10.10.0.0/16"
+}
+
+resource "aws_subnet" "test" {
+  vpc_id = "${aws_vpc.test.id}"
+  cidr_block = "10.10.0.0/21"
+  map_public_ip_on_launch = true
+  availability_zone = "us-west-2a"
+}
+
+resource "aws_api_gateway_vpc_link" "test" {
+  name = "tf-apigateway-update-%s"
+  description = "test update"
+  target_arn = "${aws_lb.test_a.arn}"
+}
+`, rName, rName)
+}
```

Comment:
We should also add a test with an actual API Gateway and a HTTP endpoint, so that we ensure it's all ok.
I'm actually wondering whether we need a hard dependency (using `depends_on`) on the HTTP integration since this resource is just creating a link between the API Gateway service & the VPC targets... 🤔 
Not sure if this could result in race dependencies! 🤷‍♂️

Reply:
Sorry, I couldn't understand what you meant so cloud you describe more details:bow: ? @Ninir

Reply:
I guess what @Ninir means is that we should add a test which tests the full setup including API Gateway integration leveraging the VPC Link.

It seems we'd also need to update other API Gateway resources to make that happen, namely to add few more attributes to `aws_api_gateway_integration` per http://docs.aws.amazon.com/apigateway/api-reference/resource/integration/#connectionType

I'm personally ok with shipping this resource as is and addressing the above in a separate PR to keep diffs small and easier to review - what do you think @Ninir ?